.github/workflows/smoke.yml

```diff
@@ -0,0 +1,62 @@
+name: Smoke tests for AWS SAM CLI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '0 5 * * *' # 05:00 server time every day
+
+# Only one pull-request triggered run should be executed at a time
+# (head_ref is only set for PR events, otherwise fallback to run_id which differs for every run).
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+env:
+  SAM_CLI_TELEMETRY: "0"
+  AWS_ACCESS_KEY_ID: "test"
+  AWS_SECRET_ACCESS_KEY: "test"
+  AWS_DEFAULT_REGION: us-east-1
+  AWS_REGION: us-east-1
+  AWS_ENDPOINT_URL: http://localhost:4566
+  VENV_DIR: ".venv"
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
```

Comment:
Can we add a timeout here so the action fails before the default 90 minutes? For example: `timeout-minutes: 5`